testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/suspend/WebSuspendTestCase.java

```diff
@@ -4,113 +4,205 @@
+ * i.e. from {@link jakarta.servlet.ServletRequestListener#requestInitialized(jakarta.servlet.ServletRequestEvent)} to {@link jakarta.servlet.ServletRequestListener#requestDestroyed(jakarta.servlet.ServletRequestEvent)},
+ * and that requests fail with the expected status code once suspended.
  */
 @RunWith(Arquillian.class)
+@RunAsClient
 public class WebSuspendTestCase {
 
-    protected static Logger log = Logger.getLogger(WebSuspendTestCase.class);
+    @Deployment
+    public static WebArchive deployment() {
+        return ShrinkWrap.create(WebArchive.class, WebSuspendTestCase.class.getSimpleName() + ".war")
+                .addPackage(SuspendStateServlet.class.getPackage())
+                ;
+    }
 
+    private static final String RUNNING = SuspensionStateProvider.State.RUNNING.toString();
+    private static final String SUSPENDING = SuspensionStateProvider.State.SUSPENDING.toString();
+    private static final Duration REQUEST_DURATION = Duration.ofMillis(TimeoutUtil.adjust(1000));
+    private static final Set<String> EVENTS = Set.of(SuspendStateServletRequestListener.INIT_EVENT_NAME, SuspendStateServlet.EVENT_NAME, SuspendStateServletRequestListener.DESTROY_EVENT_NAME);
 
+    @ArquillianResource(SuspendStateServlet.class)
+    private URL baseURL;
     @ArquillianResource
-    private ManagementClient managementClient;
+    private ManagementClient client;
 
-    @Deployment
-    public static WebArchive deployment() {
+    private final ExecutorService executor = Executors.newFixedThreadPool(1);
 
-        WebArchive war = ShrinkWrap.create(WebArchive.class, "web-suspend.war");
-        war.addPackage(WebSuspendTestCase.class.getPackage());
-        war.addPackage(HttpRequest.class.getPackage());
-        war.addClass(TestSuiteEnvironment.class);
-        war.addAsResource(new StringAsset("Dependencies: org.jboss.dmr, org.jboss.as.controller, org.jboss.remoting\n"), "META-INF/MANIFEST.MF");
-        war.addAsManifestResource(createPermissionsXmlAsset(
-                new PropertyPermission("management.address", "read"),
-                new PropertyPermission("jboss.http.port", "read"),
-                new PropertyPermission("node0", "read"),
-                // executorService.shutdown() needs the following permission
-                new RuntimePermission("modifyThread"),
-                // ManagementClient needs the following permissions and a dependency on 'org.jboss.remoting3' module
-                new RemotingPermission("createEndpoint"),
-                new RemotingPermission("connect"),
-                // HttpClient needs the following permission
-                new SocketPermission(TestSuiteEnvironment.getServerAddress() + ":" + TestSuiteEnvironment.getHttpPort(), "connect,resolve"),
-                new FilePermission(System.getProperty("jboss.inst") + "/standalone/tmp/auth/*", "read")
-        ), "permissions.xml");
```

Comment:
@pferraro Your change to using the MBean nicely eliminated all these, but you still need a createMBeanServer permission, hence the SM CI job failure.

Reply:
Fixed. SM-enabled execution should pass now.